source/server-discovery-and-monitoring/tests/unified/logging-replicaset.yml

```diff
@@ -0,0 +1,291 @@
+        messages:
+          - level: debug
+            component: topology
+            data:
+              message: "Starting topology monitoring"
+              topologyId: { $$exists: true }
+          - level: debug
+            component: topology
+            data:
+              message: "Topology description changed"
+              topologyId: { $$exists: true }
+              previousDescription:  { $$exists: true } # unknown topology
+              newDescription: { $$exists: true } # ReplicaSet topology
+          - level: debug
+            component: topology
+            data:
+              message: "Topology description changed"
+              topologyId: { $$exists: true }
+              previousDescription:  { $$exists: true }
+              newDescription: { $$exists: true } # server connected
+          - level: debug
+            component: topology
+            data:
+              message: "Topology description changed"
+              topologyId: { $$exists: true }
+              previousDescription:  { $$exists: true }
+              newDescription: { $$exists: true } # server connected
+          - level: debug
+            component: topology
+            data:
+              message: "Topology description changed"
+              topologyId: { $$exists: true }
+              previousDescription:  { $$exists: true }
+              newDescription: { $$exists: true } # server connected
+          - level: debug
+            component: topology
+            data:
+              message: "Stopped server monitoring"
+              topologyId: { $$exists: true }
+              serverHost: { $$type: string }
+              serverPort: { $$type: [int, long] }
+          - level: debug
+            component: topology
+            data:
+              message: "Stopped server monitoring"
+              topologyId: { $$exists: true }
+              serverHost: { $$type: string }
+              serverPort: { $$type: [int, long] }
+          - level: debug
+            component: topology
+            data:
+              message: "Stopped server monitoring"
+              topologyId: { $$exists: true }
+              serverHost: { $$type: string }
+              serverPort: { $$type: [int, long] }
+          - level: debug
+            component: topology
+            data:
+              message: "Topology description changed"
+              topologyId: { $$exists: true }
+              previousDescription:  { $$exists: true } # ReplicaSet topology
+              newDescription: { $$exists: true } # unknown topology
+          - level: debug
+            component: topology
+            data:
+              message: "Stopped topology monitoring"
+              topologyId: { $$exists: true }
+  - description: Successful heartbeat
+    operations:
+      - name: createEntities
+        object: testRunner
+        arguments:
+          entities:
+          - client:
+              id: *client
+              observeLogMessages:
```

Comment:
I believe the intention is that when a client is first created it should immediately start monitoring (not wait 10 seconds). Otherwise we could be adding extra latency when the client is first used.

Reply:
Yes, C# starts monitoring immediately, looks like [SDAM monitoring](https://github.com/mongodb/specifications/blob/a10e270087bc8ec661561be8cffd287bc6982d91/source/server-discovery-and-monitoring/server-monitoring.rst#monitor-thread) code behaves the same.

Reply:
The Go Driver will also begin monitoring immediately, but the server [updates](https://github.com/mongodb/mongo-go-driver/blame/master/x/mongo/driver/topology/server.go#L209) are done concurrently. This is why there is a conflict between waiting for the event and starting the connection -- i.e. it is possible that the test-runner code for awaiting events starts before monitoring. 

This is _essentially_ the command that the Go Driver will construct for non-legacy hello operations:

```js
db.adminCommand({
  "isMaster": 1,
  "maxAwaitTimeMS": 10000,
  "topologyVersion": {
    "processId": ObjectId("6488dd97c4627cd040311e50"),
    "counter": NumberLong("421")
  },
  "internalClient": {
    "minWireVersion": 17,
    "maxWireVersion": 17
  },
  "$db": "admin"
})
```

This command will always wait for 10 seconds before receiving a response from the server. So since we are awaiting the heartbeat succeeded event, this test will take minimally 10 seconds to complete. Does that sounds right?

Reply:
The first heartbeat is supposed to be the initial connection handshake. Does Go ignore that event?

Reply:
Thank you both, this was due to a bug in the Go Driver.